source/components/tabs/tab.ts

```diff
@@ -1,16 +1,36 @@
-import { Component, Optional, Inject, Input, Output, ViewChild, ContentChild, AfterViewInit, TemplateRef } from '@angular/core';
+import { Component, Optional, Inject, ElementRef, forwardRef, Input, Output, ViewChild, ContentChild, AfterViewInit, TemplateRef } from '@angular/core';
 import * as ng from 'angular';
 
-import { TabsetComponent, ITabHeader } from './tabset';
+import { TabsetComponent } from './tabset';
+import { FormComponent } from '../form/form';
 
+//TODO extend FormComponent
 @Component({
 	selector: 'rlTab',
-	template: require('./tab.html'),
+	template: require('./tab.html')
 })
 export class TabComponent {
-	header: ITabHeader;
+	@ContentChild(forwardRef(() => TabHeaderComponent))
+	header: TabHeaderComponent;
 
 	constructor() {
+	}
+}
+
+@Component({
+	selector: 'rlTabHeader',
+	template: '<ng-content></ng-content>',
+})
+export class TabHeaderComponent {

```

Comment:
Rather than getting the element ref and pulling the html out of it, we would like to use angular template refs to handle this. That allows consumers to have full access to their context when specifying tab content. The only drawback is that the selector has to become an attribute to allow for being on a template element (or using an asterisk), but it seems worth the tradeoff.
Examples:
https://github.com/RenovoSolutions/TypeScript-Angular-Components/blob/master/source/components/cards/cardHeader.template.ts
Used:
https://github.com/RenovoSolutions/TypeScript-Angular-Components/blob/master/source/components/simpleCardList/simpleCard.ts
and 
https://github.com/RenovoSolutions/TypeScript-Angular-Components/blob/master/source/components/simpleCardList/simpleCard.html#L6
For consistency, I think we should add template directives for all three (header, content, and footer). I can pair with you on this sometime if that would be helpful